### ISSUE TITLE
No way to logout from JupyterLab-hub for regular users

### ISSUE THREAD

**Issue by riazarbi** [jupyterhub/jupyterlab-hub#40]
I am using JupyterHub and JupyterLab-Hub on Ubuntu 16.04. JupyterHub starts as a system service at boot. 

`jupyterhub_config.py` is set so that JupyterHub directs users to /lab at login, as is recommended. All works fine. 

BUT only Admin Hub users have a 'Hub' top menu item so that they can click 'Logout'. Regular users can log in an use /lab, but there is no way for them to log out. 

Current workaround is to manually replace `/lab` with '/tree` in the url, get to regular old Jupyter tree and then logout from there.

**Comment by enriqueartal** [comment]
Confirmed in Fedora 27

**Comment by mj-harvey** [comment]
Also reported in https://github.com/jupyterhub/jupyterlab-hub/issues/39

**Comment by mj-harvey** [comment]
Following up on this. The Javascript console reports:

jupyterlab-hub: No configuration found.

which is evidently coming from the "activateHubExtension()" function. I've no idea where the "hub_host" option that it is unable to get should be being set. 

M

**Comment by minrk** [comment]
Very interesting that only admins have the extension! What Spawner are you using? My immediate suspicion is that there is a permission issue somewhere.

**Comment by riazarbi** [comment]
My jupyterhub_config.py has -

c.Spawner.cmd = [‘jupyter-labhub’]

Could very well be a permission issue. Some more info about my system configuration - 

My admin user corresponds to my only system sudo user, and the config.py file is situated in the sudo user’s home directory.

Jupyterhub is run by root, however, and enabled as a service on startup - it just looks at the sudo user’s config.py for config instructions.

Any suggestions? Happy to experiment - it’s a VM so I can roll it back if it messes up my setup.

**Comment by riazarbi** [comment]
Also, all install instructions were followed as sudo user. Should I be
enabling anything as a standard user? Assumed not, since Jupyterlab loads
just fine for each standard user after following install instructions as
sudo user. I can work in Jupyterlab as any standard user, I just can’t log
out when I am done.

**Comment by minrk** [comment]
How is Python installed? What is `sys.prefix`, and how was the jupyterlab extension installed? If any of these files are in your home directory, that's likely the issue. Additionally, check the permissions on `{sys.prefix}/share/jupyter/lab` and children and make sure that those files and directories are readable and listable by all users.

**Comment by mj-harvey** [comment]
HI,

My install's straight from conda (see #39). Hub runs a normal user with sudo perms for end users. And user can read all files in the install tree.

For reference, I just hacked out that check I mentioned above, putting in hard-coded values, rebuilt according to the README.md instructions and it worked fine.

**Comment by dfsnow** [comment]
Same issue here on Debian 9 with the same install method.

**Comment by riazarbi** [comment]
Refer to my earlier posts which detail my configuration. It appears that my config, which ran off a `jupyterhub_config.py` file in my sudo user's home directory, prevented regular users from spawning the jupyter-labhub extension.

I have gotten the 'Hub' tab to appear in regular user's lab browser interfaces so that regular users can log out. Steps taken - 

1. Work as root - 'sudo su' 
2. `jupyterhub_config.py`  should be moved out of the `sudo` user's home directory into /etc/jupyterhub/jupyterhub_config.py'
3. `chown root /etc/jupyterhub/jupyterhub_config.py`
4. 'chmod 0644 /etc/jupyterhub/jupyterhub_config.py`
5. If you are running `jupyterhub` as a system service, change your system service files at `/etc/systemd/system/jupyterhub.service` and `/lib/systemd/system/jupyterhub.service` to reference the `/etc/jupyterhub/jupyterhub_config.py` config, not your sudo user's config.py.
6. `jupyter labextension install @jupyterlab/hub-extension`
7. `service jupyterhub restart`

I'm sure some of these steps are unnecessary, but I am not sure which ones. 

In short, the issue was fixed by moving from a non-standard (although very common) install configuration to a more standard configuration.

**Comment by jstaf** [comment]
I don't think this issue can yet be closed - I've never seen the "hub" tab as an admin/normal user, even after following the above instructions. 

I am seeing the following in the web console, however:
```
jupyterlab-hub: No configuration found.         index.js:27
Empty string passed to getElementById().     index.js:314
```

**Comment by jstaf** [comment]
HAH - got it! I think the problem is that I'm serving `jupyterlab-hub` from a subdirectory on my webserver, in this case `/jupyter`. After doing a little digging, it looks like the `jupyterlab-hub: No configuration found.` means that there's no configuration information being passed to Jupyterlab when it starts. When it later tries to fetch this config info, it finds no value and assumes there is no "hub_prefix" for where `jupyterlab-hub` is being served from, so the jupyterlab-hub extension serves things at the wrong location. 

To fix this, I needed to create a page_config.json file for Jupyterlab. In my case, I am serving up Jupyterhub from an Anaconda installation - so the location of page_config.json should be created at: `/path/to/anaconda/share/jupyter/lab/settings/page_config.json`. 

And the contents of `page_config.json` (it looks like this also accepts a [`hub_host` parameter](https://github.com/jupyterhub/jupyterlab-hub/blob/9929330ee05eb074bc25c0469b7bf9f2aed6705a/src/index.ts#L45) if needed):
```
{
    "hub_prefix": "/jupyter"
}
```
